### PR TITLE
Lint changed files using rubocop for pull requests using GH Actions

### PR DESCRIPTION
**Pull request by caitmich**
### Summary

Instead of running `rubocop` as a step in the Codeship CI pipeline, we will use GitHub Actions to perform our rubocop check on changed files only. We will do this by comparing to `origin/develop`
If there are newly introduced rubocop warnings, the workflow run fails. If not, it passes and you are good to go! Code will not be merged without a passing rubocop step

❗️**Note:** This will need to be pushed along with a change to our current Codeship pipeline step to avoid having the pipeline fail 

Pro Sync: [https://github.com/secroots/dradis-pro/pull/1765](https://github.com/secroots/dradis-pro/pull/1765)

### Other Information

If there's anything else that's important and relevant to your pull
request, mention that information here. This could include
benchmarks, or other information.

Thanks for contributing to Dradis!

### Copyright assignment

Collaboration is difficult with commercial closed source but we want
to keep as much of the OSS ethos as possible available to users
who want to fix it themselves.

In order to unambiguously own and sell Dradis Framework commercial
products, we must have the copyright associated with the entire
codebase. Any code you create which is merged must be owned by us.
That's not us trying to be a jerks, that's just the way it works.

Please review the [CONTRIBUTING.md](https://github.com/dradis/dradis-ce/blob/master/CONTRIBUTING.md)
file for the details.

You can delete this section, but the following sentence needs to
remain in the PR's description:

> I assign all rights, including copyright, to any future Dradis
> work by myself to Security Roots.

### Check List

- [x] Added a CHANGELOG entry

### CODE REVIEW

**Comment by caitmich** [comment]
When we push this, we will need to ensure that we remove the rubocop-ci step from the Codeship pipeline, as this will take care of that step, (and that step will continuously fail bc of the reference to origin/head)

File: .github/workflows/rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Lint
+
+on:
+  pull_request:
+    branches:
+      - '*'
+
```
Comment (MattBudz) [inline .github/workflows/rubocop.yml:7]: If we only do pull_request I believe the action will only run on the first commit to the branch and on merges. Adding `push` will also run in the cop on subsequent code sommits

File: .rubocop.yml [hunk 2]
```diff
@@ -1,8 +1,9 @@
 AllCops:
-  TargetRubyVersion: 2.4
+  TargetRubyVersion: 3.0
```
Comment (MattBudz) [inline .rubocop.yml:2]: if you remove this line then rubocop will use `.ruby-version` (probably cleaner as we won't have to update this every time we bump ruby).

File: .rubocop.yml [hunk 3]
```diff
@@ -130,5 +127,4 @@ Layout/EndAlignment:
 
 # Use my_method(my_arg) not my_method( my_arg ) or my_method my_arg.
 Lint/RequireParentheses:
-  Enabled: true
-
+  Enabled: true
```
Comment (MattBudz) [inline .rubocop.yml:130]: Missing EOF new line

File: .github/workflows/rubocop.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+name: Lint
+
+on:
+  pull_request:
+    branches:
+      - '*'
```
Comment (MattBudz) [inline .github/workflows/rubocop.yml:6]:
```suggestion
on:
  pull_request:
    branches:
      - '*'
  push:
    branches:
      - '*'

```

If we only do pull_request I believe the action will only run on the first commit to the branch and on merges. Adding `push` will also run in the cop on subsequent code commits

File: .rubocop.yml (hunk 2 above)
Comment (MattBudz) [inline .rubocop.yml:2]:
```suggestion

```

if you remove this line then rubocop will use .ruby-version (probably cleaner as we won't have to update this every time we bump ruby).

File: .rubocop.yml (hunk 3 above)
Comment (MattBudz) [inline .rubocop.yml:130]:
```suggestion
  Enabled: true

```

Missing EOF new line

Comment (caitmich) [reply .rubocop.yml:129]: Thanks, I have since updated my vscode settings so it will now do it automatically!

File: .rubocop.yml (hunk 2 above)
Comment (caitmich) [reply .rubocop.yml]: Made this change, but we have to update rubocop because the current `1.13.0` doesn't support ruby version 3.1.2

**Pull request merged**